enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,295 @@
+4.x vs. 3.x.
+
+### Goals
+
+No longer require (though still allow) manual manipulation of the BuildConfig to consume entitlements.
+
+No longer require (though still allow) manual injection of subscription credentials into a user's namespace.
+
+### Non-Goals
+
+Reach a V3.x level, where no post install activity is needed to consume entitlements.
+
+In other words, no changes to the V4.x installer to allow the user to inject the subscription
+particulars.
+
+## Proposal
+
+### User Stories [optional]
+
+No new user stories are introduced here.  We are improving the user's 4.x experience for existing stories.
+
+### Implementation Details/Notes/Constraints [optional]
+
+So the concerns for the implementation center around 4 questions:
+
+- Where do we get the credentials from
+- In what form are the credentials provided (pem files, SubscriptionManager, Satellite)
+- How does the build consume the credentials
+- When does the build consume the credentials (where consuming means telling buildah to mount it)
+
+#### Where do we get the credentials from
+
+##### Current Preferred Option
+
+The most likely choice here will be to have a well known config map in the openshift-config
+namespace where administrators can store the credentials.
+ 
+The build controller can access that content and mount into build pods as it deems fit.
+
+The current assumption is that only one set of entitlement credentials is needed at the global level.
+
+##### Alternatives (can provide multiple options if deemed necessary)
+
+###### User Namespace Option
+
+Users could provide secrets/config maps with a well known annotation in their namespace that the build controller 
+can look for and mount accordingly.
+
+A slight step up from having to manually cite those secrets/config maps in the BuildConfig Spec after creating them,
+as documented today. 
+
+Whether we do this one in part may be determined by whether we expect multiple sets of entitlements to be 
+used on a given cluster, where a single global copy is not sufficient.
+
+Multiple global copies could also be an alternative here.
+
+The per namespace version would act as an override and take precedence over the global copy.
+
+###### Host Injected Option
+
+`HostPath` volumes are already called out as a security concern in [volume mounted injections for builds](volume-secrets.md)
+ proposal.
+
+The current mindset for this enhancement proposal is to also cite that concern, and only depend on such a 
+mechanism if the CRI-O/MCO based solution noted in open questions is available.
+
+#### In what form are the credentials provided
+
+There will also be well defined keys within the global config map that account for the various forms:
+
+- set of pem files
+- SubscriptionManager config and certs
+- Satellite config and instance information, yum repo file if needed, and certs  
+
+#### How does the build consume the credentials
+
+##### Current Preferred Option
```

Comment:
do we only copy at point of usage?  so if a namespace never has a build, we never copy?

Reply:
in the currently defined preferred approach, that would be correct.  We make a new/unique configmap/secret each time a build is launched.  If you never launch a build, it never happens.  (But if you launch 1000 builds, it happens 1000 times.  The secret/configmap is lifecycled with the build object).

Reply:
the performance trade off is documented in the proposal

FYI I'm leaning towards backing off of a "perferred" option based on discussion here and the broader entitlements needs that arose from the epic readouts last week ... will announce if/when I've pulled the trigger on such a change.